src/NUnitFramework/framework/Constraints/Numerics.cs

```diff
@@ -107,6 +105,7 @@ internal static bool IsFixedPointNumeric(Type type)
             {
                 if (type == typeof(byte)) return true;
                 if (type == typeof(sbyte)) return true;
+                if (type == typeof(decimal)) return true;
```

Comment:
`Decimal` is not fixed point. It's a base 10 floating-point number. (Floating point means it contains an exponent part that specifies where the decimal point goes.) `Decimal` is also correctly described this way at e.g. https://docs.microsoft.com/en-us/dotnet/api/system.decimal.

The method names should change or the caller should do something else to differentiate between the two binary floating-point types and the decimal floating-point type.

Reply:
Yeah. I'm also surprised it was not treated as an explicit floating type by these functions in nunit prior to 3.13.2

Instead of adding it here, should we remove it from the object overload of IsFixedPointNumeric in a separate PR (a breaking change).

Reply:
Oh yes, I think we should. It's a bugfix.

Reply:
Hm, actually, as I think about this more I may be able to fix the precision issue while keeping `decimal` in the floating point check and without adding this line.
This should allow the final changeset for this fix to more easily be cherry picked into v4 branch